packages/ui/src/ui/store/selectors/acl.js

```diff
@@ -60,9 +64,23 @@ function FilterBySubject(items, subjectFilter) {
     });
 }
 
+function FilterByPermissions(items, permissionsFilter) {
+    return _.filter(items, (item) => {
+        const {permissions} = item;
+        return _.difference(permissionsFilter, permissions).length === 0;
+    });
+}
+
 export const getAllObjectPermissionsFiltered = createSelector(
-    [getAllObjectPermissionsWithSplittedSubjects, getObjectSubjects],
-    FilterBySubject,
+    [
+        getAllObjectPermissionsWithSplittedSubjects,
+        getObjectSubjectFilter,
+        getObjectPermissionsFilter,
+    ],
+    (items, subjectFilter, permissionsFilter) => {
+        const buf = FilterBySubject(items, subjectFilter);
+        return FilterByPermissions(buf, permissionsFilter);
```

Comment:
It is possible to iterate over all items only once but you need to prepare a predicate function, we have some helpers in predicates.ts, please look up at `concatByAnd`. Also there is no need to filter items if filters are empty.

Reply:
fixed 👍🏻